### PR TITLE
Remove shipping method display_on mechanism

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj it looks like `#code` is a dependency of a few tests, which appears to be causing the spec failures.

File: core/db/structure.sql [hunk 1]
```diff
@@ -1361,12 +1361,9 @@ ALTER SEQUENCE spree_shipping_method_categories_id_seq OWNED BY spree_shipping_m
 CREATE TABLE spree_shipping_methods (
     id identifier NOT NULL,
     name character varying(255) NOT NULL,
-    display_on character varying(9) NOT NULL,
     deleted_at timestamp without time zone,
     created_at timestamp without time zone NOT NULL,
-    updated_at timestamp without time zone NOT NULL,
-    code character varying(255),
```
Comment (dkubb) [inline core/db/structure.sql]: This removal appears to be causing a few spec failures.

**Comment by mbj** [comment]
> @mbj it looks like #code is a dependency of a few tests, which appears to be causing the spec failures.

Yeah I'll either leave `code` in for now, or fix the specs.

**Comment by dkubb** [comment]
> Yeah I'll either leave code in for now, or fix the specs.

If leaving it in allows the specs to pass with no problems then I'd probably opt for that and do a second pass to remove `#code` rather than hold things back to combine things.

File: core/app/models/spree/shipping_method.rb [hunk 2]
```diff
@@ -15,7 +15,6 @@ class ShippingMethod < Spree::Base
                                     :foreign_key => 'shipping_method_id'
 
     validates :name,       presence: true
```
Comment (dkubb) [inline core/app/models/spree/shipping_method.rb]: Should the formatting of this line be changed given the removal on the next line?

**Comment by dkubb** [comment]
@mbj I only had one comment on this PR.

**Comment by mbj** [comment]
@dkubb Ready for the next pass?

**Comment by dkubb** [comment]
@mbj I am fine with this change. Will merge once the build is green.

**Pull request merged**